test/trento/operations/host_policy_test.exs

```diff
@@ -551,10 +565,60 @@ defmodule Trento.Operations.HostPolicyTest do
           systemd_units: [
             build(:host_systemd_unit, name: "pacemaker.service", unit_file_state: "disabled")
           ],
+          application_instances: [],
+          database_instances: [],
           cluster_id: cluster.id
         )
 
       {:error, _} = HostPolicy.authorize_operation(:reboot, host, %{})
     end
+
+    test "should authorize host reboot if all application and database instances are stopped" do
+      host =
+        build(:host,
+          cluster: nil,
+          cluster_id: nil,
+          application_instances: [
+            build(:application_instance, health: Health.unknown()),
+            build(:application_instance, health: Health.unknown())
+          ],
+          database_instances: [
+            build(:database_instance, health: Health.unknown()),
+            build(:database_instance, health: Health.unknown())
+          ]
+        )
+
+      assert :ok == HostPolicy.authorize_operation(:reboot, host, %{})
+    end
+
+    test "should forbid host reboot if not all application instances are stopped" do
+      host =
+        build(:host,
+          cluster: nil,
+          cluster_id: nil,
+          application_instances: [
+            build(:application_instance, health: Health.unknown()),
+            build(:application_instance, health: Health.passing())
+          ],
+          database_instances: []
+        )
+
+      {:error, _} = HostPolicy.authorize_operation(:reboot, host, %{})
```

Comment:
I would find nice to check the specific error message as well.

Reply:
Done it!